src/WebJobs.Script/Utility.cs

```diff
@@ -541,10 +541,10 @@ internal static void AddFunctionError(IDictionary<string, ICollection<string>> f
         {
             functionName = isFunctionShortName ? functionName : Utility.GetFunctionShortName(functionName);
 
-            ICollection<string> functionErrorCollection = new Collection<string>();
+            ICollection<string> functionErrorCollection = new HashSet<string>();
```

Comment:
I think there is a chance for some further optimization here:
1. We can avoid the extra hash set allocation and move it to be declared as part of the `out` param.
    - `!functionErrors.TryGetValue(functionName, out ICollection<string> functionErrorCollection)`
2. The check for `!string.IsNullorEmpty(functionName)` is interesting. If it is null or empty we won't even add `functionErrorCollection` to the dictionary, in which case we can just exit after line 542. But this begs the question - is null or empty even valid here, or is this indicative of a misuse of this function? And maybe we should throw an `ArgumentException` (fail fast principal)?